From c763cf351ea25b7fb31f2d388dab0838565d1b75 Mon Sep 17 00:00:00 2001
From: Dave Stevenson <dave.stevenson@raspberrypi.com>
Date: Fri, 13 Sep 2024 14:44:08 +0100
Subject: [PATCH 1262/1350] drm: panel: ili9881: Correct symmetry on
 enable/disable return codes

ili9881c_enable is always returning 0.

ili9881c_disable was returning the error code from
mipi_dsi_dcs_set_display_off.
If non-zero, the drm_panel framework will leave the panel marked as
enabled, and not run the enable hook next time around. That isn't
helpful, particularly as we're expecting unprepare to disable
resets and regulators.

Change ili9881c_disable to match enable in always returning 0.

Signed-off-by: Dave Stevenson <dave.stevenson@raspberrypi.com>
---
 drivers/gpu/drm/panel/panel-ilitek-ili9881c.c | 4 +++-
 1 file changed, 3 insertions(+), 1 deletion(-)

--- a/drivers/gpu/drm/panel/panel-ilitek-ili9881c.c
+++ b/drivers/gpu/drm/panel/panel-ilitek-ili9881c.c
@@ -1745,7 +1745,9 @@ static int ili9881c_disable(struct drm_p
 {
 	struct ili9881c *ctx = panel_to_ili9881c(panel);
 
-	return mipi_dsi_dcs_set_display_off(ctx->dsi);
+	mipi_dsi_dcs_set_display_off(ctx->dsi);
+
+	return 0;
 }
 
 static int ili9881c_unprepare(struct drm_panel *panel)
